aiida/orm/nodes/data/cif.py

```diff
@@ -196,27 +196,48 @@ def pycifrw_from_cif(datablocks, loops=None, names=None):
 
 def parse_formula(formula):
     """
-    Parses the Hill formulae, written with spaces for separators.
+    Parses the Hill formulae. Does not need spaces as separators.
+    Works also for partial occupancies and for chemical groups enclosed in round/square/curly brackets.
+    Elements are counted and a dictionary is returned.
+    e.g.  'C[NH2]3NO3'  -->  {'C': 1, 'N': 4, 'H': 6, 'O': 3}
     """
     import re
 
+    def chemcount_str_to_number(string):
+        if (string is None) or (len(string) == 0):
+            quantity = 1
+        else:
+            if re.match(r'^\d+$', string):  # check if it is an integer number
+                quantity = int(string)
+            else:
+                quantity = float(string)
```

Comment:
```suggestion
        elif re.match(r'^\d+$', string):  # check if it is an integer number
            quantity = int(string)
        else:
            quantity = float(string)
```

Reply:
One could also simply first make it a `float` and then use the `float` method `is_integer()` to determine whether it should be cast to an `int` or not.